apps/api/src/app/topics-v2/topics.controller.ts

```diff
@@ -0,0 +1,315 @@
+    summary: 'Create or update a topic',
+    description: 'Creates a new topic if it does not exist, or updates an existing topic if it already exists',
+  })
+  @ApiResponse(TopicResponseDto, 201)
+  @ApiResponse(TopicResponseDto, 200)
+  @SdkMethodName('create')
+  async upsertTopic(
+    @UserSession() user: UserSessionData,
+    @Body() body: CreateUpdateTopicRequestDto,
+    @Res({ passthrough: true }) response: Response
+  ): Promise<TopicResponseDto> {
+    const result = await this.upsertTopicUsecase.execute(
+      UpsertTopicCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        key: body.key,
+        name: body.name,
+      })
+    );
+
+    if (result.created) {
+      response.status(HttpStatus.CREATED);
+    }
+
+    return result.topic;
+  }
+
+  @Get('/:topicKey')
+  @UserAuthentication()
+  @ExternalApiAccessible()
+  @SdkMethodName('get')
+  @ApiOperation({ summary: 'Get topic by key' })
+  @ApiParam({ name: 'topicKey', description: 'The key identifier of the topic', type: String })
+  @ApiResponse(TopicResponseDto, 200)
+  async getTopic(@UserSession() user: UserSessionData, @Param('topicKey') topicKey: string): Promise<TopicResponseDto> {
+    return await this.getTopicUsecase.execute(
+      GetTopicCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        topicKey,
+      })
+    );
+  }
+
+  @Patch('/:topicKey')
+  @UserAuthentication()
+  @ExternalApiAccessible()
+  @SdkMethodName('update')
+  @ApiOperation({ summary: 'Update topic by key' })
+  @ApiParam({ name: 'topicKey', description: 'The key identifier of the topic', type: String })
+  @ApiResponse(TopicResponseDto, 200)
+  async updateTopic(
+    @UserSession() user: UserSessionData,
+    @Param('topicKey') topicKey: string,
+    @Body() body: UpdateTopicRequestDto
+  ): Promise<TopicResponseDto> {
+    return await this.updateTopicUsecase.execute(
+      UpdateTopicCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        topicKey,
+        name: body.name,
+      })
+    );
+  }
+
+  @Delete('/:topicKey')
+  @UserAuthentication()
+  @ExternalApiAccessible()
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Delete topic by key' })
+  @ApiParam({ name: 'topicKey', description: 'The key identifier of the topic', type: String })
+  @ApiResponse(DeleteTopicResponseDto, 200, false, true, {
+    description: 'Topic deleted successfully',
+  })
+  async deleteTopic(
+    @UserSession() user: UserSessionData,
+    @Param('topicKey') topicKey: string
+  ): Promise<DeleteTopicResponseDto> {
+    await this.deleteTopicUsecase.execute(
+      DeleteTopicCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        topicKey,
+        force: true,
```

Comment:
Let's fix this one, we are telling the users in the exception that they can pass `force=true` while it is not possible.

```suggestion
        force: query.force || false,
```

Reply:
We actually don't tell this to users, I have kept it as only internal thing. Where di you see it?

Reply:
I noticed this in the DeleteTopicUseCase. Just sharing a few thoughts, no need to answer directly:
If we don’t plan to expose this route externally, do we really need this logic?
If or when we do expose it, we should make sure to revisit and update this part accordingly.